File: benchmark/nodejs/factorial_recursive.js
const start = Date.now();
function fact(n) {
    if (n <= 1) return 1n;
    return BigInt(n) * fact(n - 1);
}
const result = fact(150);
const end = Date.now();
console.log(`fact(150) = ${result}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/python/factorial_recursive.py
import time

def fact(n):
    if n <= 1:
        return 1
    return n * fact(n - 1)

start = time.time() * 1000
result = fact(150)
end = time.time() * 1000
print(f"fact(150) = {result}")
print(f"Time: {end - start:.0f}ms")

